library/constant_time.c

```diff
@@ -514,6 +518,10 @@ int mbedtls_ct_hmac( mbedtls_svc_key_id_t key,
     PSA_CHK( psa_hash_update( &operation, add_data, add_data_len ) );
     PSA_CHK( psa_hash_update( &operation, data, min_data_len ) );
 
+    /* Clear the hash buffer in advance, thus stopping any access of
+     * potentially uninitialised data. */
```

Comment:
As in `mbedtls_ct_memcpy_offset`, the problem is not access to uninitialized memory, it's what `mbedtls_ct_hmac` does if `data_len_secret` is not between `min_data_len` and `max_data_len` inclusive.

I think here, it makes more sense to ensure that `output` is always set to something, and make it something that can't possibly be a valid MAC on error. So memsetting to zero makes sense.